lib/gpio/pin.toit

```diff
@@ -142,9 +143,136 @@ class Pin:
+    chip := Chip path
+    try:
+      return Pin.linux num
+          --chip=chip
+          --input=input
+          --output=output
+          --pull-up=pull-up
+          --pull-down=pull-down
+          --open-drain=open-drain
+          --initial-value=initial-value
+    finally:
+      chip.close
+
+  /**
+  Opens a GPIO Pin on the given $chip and $num (often called "offset").
+
+  This constructor only works on Linux.
+
+  See $(constructor num) for more information on the remaining parameters.
+  */
+  constructor.linux num/int
+      --chip/Chip
+      --input/bool=false
+      --output/bool=false
+      --pull-up/bool=false
+      --pull-down/bool=false
+      --open-drain/bool=false
+      --initial-value/int=0:
+    return PinLinux_ num
+        --chip=chip
+        --input=input
+        --output=output
+        --pull-up=pull-up
+        --pull-down=pull-down
+        --open-drain=open-drain
+        --initial-value=initial-value
+
+  /**
+  Opens a GPIO pin based on the given $name.
+
+  This constructor only works on Linux.
+
+  If a path to a chip is provided, finds the pin on that chip. Otherwise, searches
+    all available chips (see $Chip.list).
+  Names are not guaranteed to be unique. If multiple pins have the same name, the
+    first pin found is returned.
+
+  See $(constructor num) for more information on the remaining parameters.
+  */
+  constructor.linux
+      --name/string
+      --path/string?=null
+      --input/bool=false
+      --output/bool=false
+      --pull-up/bool=false
+      --pull-down/bool=false
+      --open-drain/bool=false
+      --initial-value/int=0:
+
+    try-chip := : | path/string |
+      chip := Chip path
+      try:
+        offset := gpio-linux-chip-offset-for-name_ chip.resource_ name
+        if offset != -1:
+          return Pin.linux offset
+              --chip=chip
+              --input=input
+              --output=output
+              --pull-up=pull-up
+              --pull-down=pull-down
+              --open-drain=open-drain
+              --initial-value=initial-value
+      finally:
+        chip.close
+
+    if path:
+      try-chip.call path
+    else:
+      Chip.list.do try-chip
+    throw "NOT_FOUND"
+
+  /**
+  Opens a GPIO pin based on the given $name on the given $chip.
+
+  This constructor only works on Linux.
+
+  If a chip is provided, finds the pin on that chip. Otherwise, searches all available
+    chips (see $Chip.list).
+  Names are not guaranteed to be unique. If multiple pins have the same name, the
+    first pin found is returned.
+
+  See $(constructor num) for more information on the remaining parameters.
+  */
+  constructor.linux
+      --name/string
+      --chip/Chip
+      --input/bool=false
+      --output/bool=false
+      --pull-up/bool=false
+      --pull-down/bool=false
+      --open-drain/bool=false
+      --initial-value/int=0:
+
+    offset := gpio-linux-chip-offset-for-name_ chip.resource_ name
+    if offset == -1: throw "NOT_FOUND"
```

Comment:
I often end up just using the sign. The magic value -1 seems worse than just using the sign bit. YMMV.

```suggestion
    if offset < 0: throw "NOT_FOUND"
```

Reply:
done.